com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/ShaderProgramBuilderTest.java

```diff
@@ -138,4 +138,46 @@ public void testShaderPrograms() throws Exception {
         GetProject().getProjectProperties().putBooleanValue("shader", "output_spirv", true);
         doTest(true);
     }
+
+    private void testOutput(String expected, String source) {
+        if (!expected.equals(source)) {
+            System.err.printf("EXPECTED:\n'%s'\n", expected);
+            System.err.printf("SOURCE:\n'%s'\n", source);
+        }
+        assertEquals(expected, source);
+    }
+
+    @Test
+    public void testGlslDirectives() throws Exception {
+        String source;
+        String expected;
+
+        source = ShaderProgramBuilder.compileGLSL("", ShaderUtil.ES2ToES3Converter.ShaderType.VERTEX_SHADER, ShaderDesc.Language.LANGUAGE_GLSL, "", "", true);
+        expected =  "#version 140\n" +
+                    "#ifndef GL_ES\n" +
+                    "#define lowp\n" +
+                    "#define mediump\n" +
+                    "#define highp\n" +
+                    "#endif\n" +
+                    "\n" +
+                    "#line 0\n";
+        testOutput(expected, source);
+
+        source = "#extension GL_OES_standard_derivatives : enable\n" +
+                 "varying highp vec2 var_texcoord0;";
```

Comment:
Basically, we need to keep the lines starting with "#" at the top of the resulting file (Although "#version" always is at the absolute top).